acceptance/tests/ensure_facter_3_and_facter_4_output_matches.rb

```diff
@@ -1,57 +1,52 @@
-                    processors\.physicalcount selinux swapfree_mb swapsize_mb system_uptime\.days system_uptime\.hours
-                    system_uptime\.seconds uptime_days uptime_hours uptime_seconds clientnoop swapfree
-                    disks\..*\.size_bytes hypervisors\.lpar\.partition_number mountpoints\..*\.capacity
-                    processors\.speed serialnumber hypervisors\.xen\.privileged os\.release.\minor
-                    operatingsystemrelease os\.release\.full os\.distro\.description filesystems
-                    sp_uptime system_profiler\.uptime os\.release\.minor
-                    hypervisors\.zone\..* system_uptime\.uptime uptime hypervisors\.ldom\..* ldom_.*
-                    boardassettag dmi\.board\.asset_tag is_virtual kernelmajversion lsbmajdistrelease zones virtual
-                    blockdevice_.*_vendor blockdevice_.*_size]
+  exclude_list = %w{mountpoints\..*
+    partitions\..*\.filesystem
+    partitions\..*\.size_bytes
+    partitions\..*\.mount        	
+    partitions\..*\.uuid
+    ldom_.*                   	
+    boardassettag 	
+    dmi\.board\.asset_tag 	
+    is_virtual 	
+    kernelmajversion 	
+    lsbmajdistrelease 	
+    zones 	
+    virtual                   	
+    blockdevice_.*_vendor blockdevice_.*_size
+    hypervisors.vmware.version
+    os\.distro\.description  }
 
   agents.each do |agent|
     exclude_list += ['macosx_productversion.*', 'os.macosx.version'] if agent.platform =~ /^osx-11/
 
-    teardown do
-      step 'disable facterng feature flag' do
-        on agent, puppet('config set facterng false')
+    step 'run puppet facts diff ' do
+      on agent, puppet('facts diff') do
+        @diff = stdout
+        
       end
     end
 
-    step 'obtain output from puppet facts running with Facter 3.x' do
-      on agent, puppet('facts') do
-        @facter3_output = stdout
-      end
-    end
-
-    step 'enable facterng feature flag' do
-      on agent, puppet('config set facterng true')
-    end
-
-    step 'obtain output from puppet facts running with Facter 4.x' do
-      on agent, puppet('facts') do
-        @facter4_output = stdout
+    step 'build exclude list' do
+      diff_hash = JSON.parse(@diff)
+      diff_hash.each do |key, value|
+        unless value['old_value'] != nil
+          exclude_list.append(key)
+        end
       end
     end
 
     step 'compare Facter 3 to Facter 4 outputs' do
-      fact_dif = FactDif.new(@facter3_output, @facter4_output, exclude_list)
-      unless fact_dif.difs.empty?
-        fail_test("Facter 3 and Facter 4 outputs have the fallowing differences:  #{fact_dif.difs}")
+      ignored_facts = exclude_list.join("|")
+      on(agent, puppet("facts diff --exclude '#{ignored_facts}'")) do 
+        diff = JSON.parse(stdout)
+        unless diff.size.zero?
+          fail_test("Facter 3 and Facter 4 outputs have the following differences:  #{stdout}")
+        end
       end
     end
   end
-end
+end
```

Comment:
Rather then executing `puppet facts diff` and building a new exclude list, then execute `puppet facts diff` again,  I suggest to run `puppet facts diff` once, with a known exclude list and parse the output.
exclude_list - contains all the known differences, and should not make this test fail.

Something like:
```ruby
on agent, puppet("facts diff '#{exclude_list}'") do |output|
  diff = JSON.parse(output

  # rep_diff is the diff that we are interested in, and should make this test to fail
  # this ignores new facts added only in Facter 4
  rep_diff = diff.delete_if {|k, v| v['old_value'] == 'null' }
  fail_test if rep_diff.keys.size.positive?
end